src/Integration.Vsix/Delegates/SonarAnalyzerConnectedWorkflow.cs

```diff
@@ -60,53 +59,40 @@ internal class SonarAnalyzerConnectedWorkflow : SonarAnalyzerWorkflowBase
             this.boundProject = boundProject;
             this.suppressionHandler = suppressionHandler;
 
-            SonarAnalysisContext.ShouldExecuteRuleFunc = ShouldExecuteVsixAnalyzer;
-            SonarAnalysisContext.ReportDiagnosticAction = VsixAnalyzerReportDiagnostic;
+            SonarAnalysisContext.ReportDiagnostic = VsixAnalyzerReportDiagnostic;
         }
 
-        internal /* for testing purposes */ bool ShouldExecuteVsixAnalyzer(IAnalysisRunContext context)
-        {
-            if (context.SyntaxTree == null)
-            {
-                return false;
-            }
-
-            Debug.Assert(context.SupportedDiagnostics?.Any() ?? false,
-                "Not expecting a null or empty collection of diagnostic descriptors");
-
-            // Disable the VSIX analyzer as we want the NuGet analyzer to take precedence
-            if (GetProjectNuGetAnalyzerStatus(context.SyntaxTree) != ProjectAnalyzerStatus.NoAnalyzer)
-            {
-                return false;
-            }
-
-            return HasAnyRuleEnabled(context.SyntaxTree, context.SupportedDiagnostics);
-        }
-
-        private bool HasAnyRuleEnabled(SyntaxTree syntaxTree, IEnumerable<DiagnosticDescriptor> supportedDescriptors) =>
-            this.qualityProfileProvider.GetQualityProfile(this.boundProject, GetLanguage(syntaxTree))
+        internal /* for testing purposes */ protected override bool? ShouldRegisterContextAction(
+            IEnumerable<DiagnosticDescriptor> descriptors) =>
+            this.qualityProfileProvider.GetQualityProfile(this.boundProject, GetLanguage(descriptors))
                 ?.Rules
                 .Select(x => x.Key)
-                .Intersect(supportedDescriptors.Select(d => d.Id)) // We assume that a rule is enabled if present in the QP
-                .Any()
-            // Fallback using SonarWay
-            ?? supportedDescriptors.Any(d => d.CustomTags.Contains(DiagnosticTagsHelper.SonarWayTag));
+                .Intersect(descriptors.Select(d => d.Id)) // We assume that a rule is enabled if present in the QP
+                .Any();
 
-        internal /* for testing purposes */ virtual Language GetLanguage(SyntaxTree syntaxTree)
+        internal /* for testing purposes */ static Language GetLanguage(IEnumerable<DiagnosticDescriptor> descriptors)
         {
-            var rootNode = syntaxTree?.GetRoot();
+            // We assume that all descriptors share the same language so we can take only the first one
```

Comment:
I wanted to put a `Debug.Assert` but the check is too complex and requires a full `#if DEBUG` section so I decided not to implement it.